app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryViewModel.kt

```diff
@@ -43,15 +44,21 @@ class PromotedStoryViewModel(
     )
   }
 
+  // Overriding equals is needed so that DataProvider combine functions used in the HomeViewModel
+  // will only rebind when the actual data in the data list changes, rather than when the ViewModel
+  // object changes.
   override fun equals(other: Any?): Boolean {
-    if (this === other) {
-      return true
-    }
-    if (other == null || other.javaClass != javaClass) {
-      return false
-    }
-    val otherResult = other as PromotedStoryViewModel
-    return otherResult.internalProfileId == this.internalProfileId &&
-      otherResult.promotedStory == this.promotedStory
+    return other is PromotedStoryViewModel &&
+      other.internalProfileId == this.internalProfileId &&
+      other.totalStoryCount == this.totalStoryCount &&
+      other.entityType == this.entityType &&
+      other.promotedStory == this.promotedStory
+  }
+
+  override fun hashCode(): Int {
+    return Objects.hash(this.internalProfileId) +
```

Comment:
Ditto here & elsewhere for variadic hash code (which is actually preferred since addition doesn't make as good of a hash as other methods)--addition can lead to more collisions which would technically hurt performance if using a hash-based data structure.

Reply:
Done, makes sense